File: submissions/0083-remove-duplicates-from-sorted-list/solution.ts
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     val: number
 *     next: ListNode | null
 *     constructor(val?: number, next?: ListNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.next = (next===undefined ? null : next)
 *     }
 * }
 */

function deleteDuplicates(head: ListNode | null): ListNode | null {
   let dummy = head;
   
   if(dummy === null){
       return null;
   }
   
    while(dummy && dummy.next){
        if(dummy.val === dummy.next.val){
            dummy.next = dummy.next.next
        } else{
            dummy = dummy.next;
        }
    }
    
    return head;
};

File: submissions/0083-remove-duplicates-from-sorted-list/solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

# 1 -> 2
# curr next
class Solution:
    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:
        dummy = head
        
        while head and head.next:
            curr = head
            next_node = head.next
            
            if curr.val == next_node.val:
                curr.next = next_node.next
            else:
                curr = curr.next 
            
            head = curr
        
        return dummy
